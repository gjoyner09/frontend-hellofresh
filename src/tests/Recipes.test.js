import { api } from '../data/index';
import renderer from 'react-test-renderer'
import AllRecipes, { Recipes } from '../components/Recipes';

jest.mock('axios', () => {
  return {
    create: jest.fn(() => ({
      get: jest.fn(),
      interceptors: {
        request: { use: jest.fn(), eject: jest.fn() },
        response: { use: jest.fn(), eject: jest.fn() }
      }
    }))
  }
})


test('should fetch all recipes', () => {
  const recipes = [{
      title: "TestTitle",
      ingredients: "TestIngredients",
      instructions: "TestInstructions",
      nutritional_info: "TestNutritionalInfo",
      classification: "TestClassification"
    }];
  const resp = {data: recipes};
  api.get.mockResolvedValue(resp);

  // or you could use the following depending on your use case:
  // axios.get.mockImplementation(() => Promise.resolve(resp))

  return AllRecipes.all().then(data => expect(data).toEqual(recipes));
})

test('Recipes page renders correctly', () => {
  const component = renderer.create(
      <Recipes />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})