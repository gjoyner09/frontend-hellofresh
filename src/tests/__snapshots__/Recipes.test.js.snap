// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes page renders correctly 1`] = `
Array [
  <h1>
    Recipes
  </h1>,
  <form
    onSubmit={[Function]}
  >
    <input
      id="title"
      label="Title"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="title"
    >
      Title
    </label>
    <br />
    <input
      id="ingredients"
      label="Ingedients"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="ingredients"
    >
      Ingredients
    </label>
    <br />
    <input
      id="instructions"
      label="Instructions"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="instructions"
    >
      Instructions
    </label>
    <br />
    <input
      id="nutritional_info"
      label="Nutritional Info"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="nutritional_info"
    >
      Nutritional Info
    </label>
    <br />
    <input
      id="classification"
      label="Classification"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="classification"
    >
      Classification
    </label>
    <br />
    <button
      type="submit"
    >
      Add recipe
    </button>
  </form>,
]
`;
